worklist/views.py

```diff
@@ -211,3 +182,22 @@ def update_task_info(request):
     results.update(progress=progress, status=status_code, claimed_by=claimed_by)
 
     return JsonResponse({'success': True})
+
+
+# View to display worklists created or being worked upon by the logged-in user
+@login_required
+def show_user_worklists(request):
+    username = request.user.username
+
+    def worklist_to_dict(worklist):
+        return {'name': worklist.name, 'tags': worklist.tags,
+                'description': worklist.description, 'psid': worklist.psid,
+                'created_by': worklist.created_by}
+
+    created_worklists = \
+        map(worklist_to_dict, WorkList.objects.filter(created_by=username))
+    worked_upon_worklists = \
+        map(worklist_to_dict, [r.worklist for r in Task.objects.filter(claimed_by=username)])
+
+    return render(request, 'user-worklists.html', {'created_worklists': list(created_worklists),
```

Comment:
Can't you just pass an iterator to `render`, without calling `list`?

Reply:
On doing that a map object will be passed to the javascript. In order to avoid that, I passed a list.